airbyte_cdk/cli/airbyte_cdk/_secrets.py

```diff
@@ -272,11 +313,37 @@ def _write_secret_file(
     secret: "Secret",  # type: ignore
     client: "secretmanager.SecretManagerServiceClient",  # type: ignore
     file_path: Path,
-) -> None:
-    version_name = f"{secret.name}/versions/latest"
-    response = client.access_secret_version(name=version_name)
+) -> Optional[str]:
+    """Write the most recent enabled version of a secret to a file.
+
+    Lists all enabled versions of the secret and selects the most recent one.
+    Returns an error message if no enabled versions are found.
+
+    Args:
+        secret: The secret to write to a file
+        client: The Secret Manager client
+        file_path: The path to write the secret to
+
+    Returns:
+        Optional[str]: Error message if no enabled version is found, None otherwise
+    """
+    # List all enabled versions of the secret
+    response = client.list_secret_versions(
+        request={"parent": secret.name, "filter": "state:ENABLED"}
+    )
+
+    versions = list(response)
+
+    if not versions:
+        secret_name = secret.name.split("/secrets/")[-1]  # Removes project prefix
+        return f"No enabled version found for secret: {secret_name}"
+
+    enabled_version = versions[0]
```

Comment:
If two versions of a secret are enabled, will this get the latest version of the oldest, or an arbitrary version?